src/client/dfs/dfs_sys.c

```diff
@@ -455,24 +453,26 @@ dfs_sys_umount(dfs_sys_t *dfs_sys)
 			d_hash_rec_decref(dfs_sys->hash, rlink);
 		}
 
-		hash_rc = d_hash_table_destroy(dfs_sys->hash, false);
-		if (hash_rc != 0) {
-			D_DEBUG(DB_TRACE, "failed to destroy hash table: "
-				DF_RC"\n", DP_RC(hash_rc));
+		rc = d_hash_table_destroy(dfs_sys->hash, false);
+		if (rc) {
+			D_DEBUG(DB_TRACE, "failed to destroy hash table: "DF_RC"\n", DP_RC(rc));
```

Comment:
The original idea with this API was to not flood the user with redundant error messages, since these functions are basically just wrappers. I.e. either the lower-level functions or the caller would be responsible for appropriate logging. This was mostly for DFS, since DFS already logs errors where appropriate, but the concept ended up being applied to all functions. There actually aren't ANY D_ERROR calls in this file. If needed, I think we should re-consider the file as a whole.